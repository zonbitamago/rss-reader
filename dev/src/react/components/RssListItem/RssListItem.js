import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./RssListItem.css";
import {
  ListItem,
  ListItemText,
  ListItemSecondaryAction
} from "material-ui/List";
import IconButton from "material-ui/IconButton";
import Delete from "@material-ui/icons/Delete";

class RssListItem extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="RssListItem">
        <ListItem component="a" href={this.props.url}>
          <ListItemText
            className="RssListItem-Text"
            primary={this.props.text}
          />
          <ListItemSecondaryAction>
            <IconButton aria-label="Delete">
              <Delete />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      </div>
    );
  }
}

RssListItem.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

RssListItem.defaultProps = {};

export default RssListItem;
