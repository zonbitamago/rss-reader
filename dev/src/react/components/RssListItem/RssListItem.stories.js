import React from "react";
import { storiesOf } from "@storybook/react";

import RssListItem from "./RssListItem";
import List from "material-ui/List";

storiesOf("Atoms/RssListItem", module)
  .addDecorator(story => <List>{story()}</List>)
  .add("RssListItem", () => (
    <RssListItem text="sample" url="https://www.google.co.jp/" />
  ));
