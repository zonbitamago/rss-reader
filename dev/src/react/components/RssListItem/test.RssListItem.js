import React from "react";
import { shallow } from "enzyme";

import RssListItem from "./RssListItem";

describe("RssListItem", () => {
  let component, props;

  beforeEach(() => {
    props = { text: "test", url: "https://google/co.jp" };
    component = shallow(<RssListItem {...props} />);
  });

  it("should", () => {
    expect(component).toMatchSnapshot();
  });
});
