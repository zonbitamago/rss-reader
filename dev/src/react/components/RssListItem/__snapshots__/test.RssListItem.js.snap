// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RssListItem should 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RssListItem
    text="test"
    url="https://google/co.jp"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(ListItem)
        component="a"
        href="https://google/co.jp"
      >
        <WithStyles(ListItemText)
          className="RssListItem-Text"
          primary="test"
        />
        <WithStyles(ListItemSecondaryAction)>
          <WithStyles(IconButton)
            aria-label="Delete"
          >
            <pure(Delete) />
          </WithStyles(IconButton)>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>,
      "className": "RssListItem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(ListItemText)
            className="RssListItem-Text"
            primary="test"
          />,
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>,
        ],
        "component": "a",
        "href": "https://google/co.jp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "RssListItem-Text",
            "primary": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(IconButton)
              aria-label="Delete"
            >
              <pure(Delete) />
            </WithStyles(IconButton)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <pure(Delete) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(ListItem)
          component="a"
          href="https://google/co.jp"
        >
          <WithStyles(ListItemText)
            className="RssListItem-Text"
            primary="test"
          />
          <WithStyles(ListItemSecondaryAction)>
            <WithStyles(IconButton)
              aria-label="Delete"
            >
              <pure(Delete) />
            </WithStyles(IconButton)>
          </WithStyles(ListItemSecondaryAction)>
        </WithStyles(ListItem)>,
        "className": "RssListItem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItemText)
              className="RssListItem-Text"
              primary="test"
            />,
            <WithStyles(ListItemSecondaryAction)>
              <WithStyles(IconButton)
                aria-label="Delete"
              >
                <pure(Delete) />
              </WithStyles(IconButton)>
            </WithStyles(ListItemSecondaryAction)>,
          ],
          "component": "a",
          "href": "https://google/co.jp",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "RssListItem-Text",
              "primary": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                aria-label="Delete"
              >
                <pure(Delete) />
              </WithStyles(IconButton)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <pure(Delete) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
